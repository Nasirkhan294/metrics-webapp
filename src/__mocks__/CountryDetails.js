import React from 'react';

const CountryDetails = () => (
  <div>
    <div>
      <img src="img" alt="country-flag" />
      <h2>
        States by country:
        <span>Pakistan</span>
      </h2>
    </div>
    <div>
      <li>
        Sub region: Asia
      </li>
      <li>
        Capital: Islamabad
      </li>
      <li>
        Population:220892331
      </li>
      <li>
        Time zone: UTC+05:00
      </li>
      <li>
        Calling Codes: 92
      </li>
    </div>
  </div>
);

export default CountryDetails;
