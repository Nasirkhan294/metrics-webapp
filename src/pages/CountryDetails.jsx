import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import styles from '../styles/CountryDetails.module.css';

const CountryDetails = () => {
  const { name } = useParams();
  const countries = useSelector((state) => state.countries.countries);
  const country = countries.find((country) => country.name === name);

  if (!country) {
    return <p>Country not found</p>;
  }

  return (
    <>
      <div className={styles.banner}>
        <img src={country.flag} alt={country.name} />
        <h2 className={styles.header}>
          States by country:
          <span>{country.name}</span>
        </h2>
      </div>
      <div>
        <li className={styles.info}>
          <p>Sub region:</p>
          {country.subregion}
        </li>
        <li className={styles.info}>
          <p>Capital:</p>
          {country.capital}
        </li>
        <li className={styles.info}>
          <p>Population:</p>
          {country.population}
        </li>
        <li className={styles.info}>
          <p>Native name:</p>
          {country.nativeName}
        </li>
        <li className={styles.info}>
          <p>Time zone:</p>
          {country.timezones}
        </li>
        <li className={styles.info}>
          <p>Calling Codes:</p>
          {country.callingCodes}
        </li>
        <li className={styles.info}>
          <p>Area:</p>
          {country.area}
        </li>
        <li className={styles.info}>
          <p>Alpha3Code:</p>
          {country.alpha3Code}
        </li>
      </div>
    </>
  );
};

export default CountryDetails;
