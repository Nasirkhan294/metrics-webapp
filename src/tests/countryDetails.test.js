import React from 'react';
import renderer from 'react-test-renderer';
import CountryDetails from '../__mocks__/CountryDetails';

describe('Country Details', () => {
  it('renders correctly', () => {
    const component = renderer.create(<CountryDetails />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
