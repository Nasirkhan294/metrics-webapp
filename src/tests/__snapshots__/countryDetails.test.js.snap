// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Details renders correctly 1`] = `
<div>
  <div>
    <img
      alt="country-flag"
      src="img"
    />
    <h2>
      States by country:
      <span>
        Pakistan
      </span>
    </h2>
  </div>
  <div>
    <li>
      Sub region: Asia
    </li>
    <li>
      Capital: Islamabad
    </li>
    <li>
      Population:220892331
    </li>
    <li>
      Time zone: UTC+05:00
    </li>
    <li>
      Calling Codes: 92
    </li>
  </div>
</div>
`;
